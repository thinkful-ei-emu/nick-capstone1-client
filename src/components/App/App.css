* {
  box-sizing: border-box;
}

body {
  background-image: url(http://vater.s3.amazonaws.com/artists/Abe%20Cunningham%20Photo%20Credit%20Ed%20Davis.jpg)
}

select:hover{
  cursor: pointer;
}

button:hover{
  cursor: pointer;
}