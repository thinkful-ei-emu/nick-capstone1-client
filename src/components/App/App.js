import React from 'react';
import { Switch, Route } from 'react-router-dom';
import PrivateRoute from '../../Utils/PrivateRoute';
import PublicOnlyRoute from '../../Utils/PublicOnlyRoute';
import LandingPage from '../../routes/LandingPage/LandingPage';
import SignupPage from '../../routes/SignupPage/SignupPage'
import PostListPage from '../../routes/PostListPage/PostListPage'
import LoginPage from '../../routes/LoginPage/LoginPage';
import PostFormPage from '../../routes/PostFormPage/PostFormPage';
import PostPage from '../../routes/PostPage/PostPage';
import './App.css';



class App extends React.Component {
  
  state = {
    hasError: false
  }
  


  render(){
    
    
    return (

      <div className="App">
        
          
        
        <Switch>
          <PublicOnlyRoute 
          exact path={'/'}
          component={LandingPage}
          />
          <Route
          exact path={'/posts'}
          component={PostListPage}
          />
          <PublicOnlyRoute
          exact path={'/login'}
          component={LoginPage}
          />
          <PublicOnlyRoute
          exact path={'/signup'}
          component={SignupPage}
          />
          <PrivateRoute
            exact path={'/postform'}
            component={PostFormPage}
          />
          <PrivateRoute
            path={'/posts/:postId'}
            component={PostPage}
            />
        </Switch>
      </div>
    );
  }
}

export default App;
