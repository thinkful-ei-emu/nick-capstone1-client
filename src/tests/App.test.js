import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App/App';
import {MemoryRouter} from 'react-router';
import renderer from 'react-test-renderer';



it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<MemoryRouter><App /></MemoryRouter>, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders UI as expected', () => {
  const tree = renderer
      .create(<MemoryRouter><App /></MemoryRouter>)
      .toJSON();
      expect(tree).toMatchSnapshot();
})

