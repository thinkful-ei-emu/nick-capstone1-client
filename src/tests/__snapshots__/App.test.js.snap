// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<div
  className="App"
>
  <div
    className="landing-page"
  >
    <h1
      className="lp-header"
    >
      BandBridge
    </h1>
    <section
      className="lp-info"
    >
      <h2>
        Find your band
      </h2>
      <p
        className="bb-desc"
      >
        BandBridge helps connect you with artists that share your musical interests. Make an account to join the conversation or continue as a guest to check it out.
      </p>
    </section>
    <section
      className="lp-links-container"
    >
      <div
        className="lp-links"
      >
        <a
          href="/signup"
          onClick={[Function]}
        >
          <button
            className="lp-button"
          >
            Sign up
          </button>
        </a>
        <a
          href="/login"
          onClick={[Function]}
        >
          <button
            className="lp-button"
          >
            Login
          </button>
        </a>
        <a
          href="/posts"
          onClick={[Function]}
        >
          <button
            className="lp-button"
          >
            Guest
          </button>
        </a>
      </div>
    </section>
  </div>
</div>
`;
